### DISCUSSION TITLE
Animation in AR Mode

### DISCUSSION THREAD

**Discussion by burungiu** [google/model-viewer#3042] [Q&A]
Hello, guys

There is a way to run GLTF/GLB animations in AR mode? I tried it on iPhone and it doesn't work. 
Also, how can I replace object directly from Quick Look? Or it's a separately instance which receive only one object?

Thank you!

**Comment by elalish** [comment]
To get consistency in iOS AR, we're still waiting for Safari to support WebXR. Until then these more advanced features only work on Android because Quick Look is a native app that we can't control from the web and  it  doesn't even  support glTF. Our  auto-generation of USDZ doesn't yet handle animation. You can however use another tool to convert to USDZ and then  set that as the `ios-src`. It just means you need to serve two models.

**Comment by jonalex** [comment]
I tried by inputing a USDZ in the `ios-src`.
On android, I have access to control the animation with `modelViewer.play()` [link](https://modelviewer.dev/examples/animation/#controlSpeed) , but on iOS the animation attached to the USDZ, which is a converter version of the GLB, will just loop out of control.  

Any idea on how to control it, even to stop it when `<model-viewer>` has loaded?

**Reply by jonalex** [reply]
Does the `<model-viewer>` object remain on the iOS version or is it converted to another object?  That would explain why I can't get the object or add any listers to it.

**Reply by elalish** [reply]
Again, since iOS doesn't support WebXR, their only AR experience (Quick Look) is an app that's outside of the browser, so indeed there is zero control there. @grorg who might have the power to change this.

**Reply by yerevin** [reply]
I would love iOS to support WebXR, there are so many problems related to differences between GLB and USDZ :(
Are there any public sites like https://expo.canny.io/feature-requests where it is possible to bring attention to that feature request to Apple?

**Reply by elalish** [reply]
You can file a bug on WebKit, which is the only public way I know of to feature request into Safari.

**Comment by burungiu** [comment]
So how exactly this works with animation on Safari on iOS?

 https://shevchenkotaras.adv.ua/?fbclid=IwAR1-VWCLhLccMT-mz2jWoVFZNY1yihQMYsMsl4t0f1VRepV3HNfu4CAhdqk

**Reply by elalish** [reply]
You can use `ios-src` to specify your own USDZ that is animated appropriately. Apple's gltf->usdz conversion can do more than ours can.

**Reply by allenwixted** [reply]
How do you recommend this for models that use material variants?

**Reply by elalish** [reply]
Good luck. Without WebXR support on iOS, there's not much I can do to enable this. (FYI @grorg).
